### DISCUSSION TITLE
How to run prettier check ?

### DISCUSSION THREAD

**Discussion by TheFlyingImagination** [alshedivat/al-folio#2623] [Q&A] [answered]
I can't seem to be able to run prettier check. I have the site hosted on github pages and i edit the files in VS code and push the changes to the repo, but i can't seem to know how to run prettier. So it always throws the exit with error code 1. 
![image](https://github.com/user-attachments/assets/9178bb4c-dfe5-41ac-b268-aa4e62a4a85c)
The help post in FAQs is mostly about how to install it. Do i need to host the site with docker in order to run prettier? Thanks in advance.

**Comment by george-gca** [comment]
Actually the installation part is to run your page locally. Running locally with docker already includes prettier, so it is the easier way.

If you don't use docker it is simple to [integrate it](https://prettier.io/docs/en/editors.html) with your preferred IDE using an extension.

Now, if you want to run it manually, you can follow the first 2 steps in [this tutorial](https://george-gca.github.io/blog/2023/slidev_for_non_web_devs/), more specifically:
- Installing node version manager (nvm)
- Installing Node (latest version)

Then, you can install it using `npm install prettier` inside the project directory, or install it globally on your computer using `npm install -g prettier`. To run `prettier` on your current directory use `npx prettier . --write`.

Another option (if you want to just check what went wrong) is to check the failed action in your GitHub repo. If you open the failed action, you can see that right after the failed step inside the action there are 2 steps:
- Create diff
- Upload html diff

The second one creates a zip with an HTML inside that contains the changes that need to be made.

![image](https://github.com/user-attachments/assets/3e644cda-6e1d-4ef8-a715-eebb818dcca6)

**Reply by Pats-Chen** [reply, marked as answer]
"Then, you can install it using ```npm install prettier``` inside the project directory, or install it globally on your computer using ```npm install -g prettier```. To run prettier on your current directory use ```npx prettier . --write```."
Works for me. Thanks a lot!

**Reply by ykchong45** [reply]
Thanks! Followed your descriptio I installed it into the docker environment, `docker compose exec jekyll bash`.
